lms/djangoapps/instructor_task/tasks_helper/misc.py

```diff
@@ -466,3 +467,62 @@ def upload_ora2_submission_files(
     TASK_LOG.info('%s, Task type: %s, Upload complete.', task_info_string, action_name)
 
     return UPDATE_STATUS_SUCCEEDED
+
+
+def generate_anon_ids(_xmodule_instance_args, _entry_id, course_id, task_input, action_name):  # lint-amnesty, pylint: disable=too-many-statements
+    """
+    Generate a 2-column CSV output of user-id, anonymized-user-id
+    """
+
+    start_time = time()
+    start_date = datetime.now(UTC)
+
+    fmt = 'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = fmt.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=task_input
+    )
+    TASK_LOG.info('%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    curr_step = {'step': "Collecting ids"}
+    TASK_LOG.info(
+        '%s, Task type: %s, Current step: %s for all learners',
+        task_info_string,
+        action_name,
+        curr_step,
+    )
+
+    students = User.objects.filter(
+        courseenrollment__course_id=course_id,
+    ).order_by('id')
+
+    task_progress = TaskProgress(action_name, students.count, start_time)
+    task_progress.update_task_state(extra_meta=curr_step)
+
+    curr_step = {'step': "Started Computing rows"}
+    TASK_LOG.info(
```

Comment:
the only thing changing is the current step -- can we extract it into a sub-method for logging information?

Reply:
we need it to update the task_progress as well.

Reply:
Sure, but keeping Logs in a single place is easy to maintain and change in the future.

Reply:
```
curr_step = {'step': "Collecting ids"}
log_and_update_progress(curr_step)
...
...
...
curr_step = {'step': "Started Computing rows"}
log_and_update_progress(curr_step)

```